linera-chain/src/manager.rs

```diff
@@ -394,36 +398,45 @@ impl ChainManager {
         Ok(Outcome::Accept)
     }
 
+    /// If the validated block certificate is more recent, returns that certificate
+    /// so that the locked block can be updated.
+    pub fn validated_block_if_newer_than_locked(
+        &self,
+        validated_block_certificate: Option<LiteCertificate<'static>>,
+        executed_block: ExecutedBlock,
+    ) -> Option<Certificate> {
+        let lite_cert = validated_block_certificate?;
+        if self
+            .locked
+            .as_ref()
+            .map_or(true, |locked| locked.round < lite_cert.round)
```

Comment:
There's no "connection" between `self.locked`'s round and the `executed_block` argument but we treat it as if they were - if locked's round number is lower than that on certificate, we put `executed_block` into the `lite_cert`. What if `locked` is for a completely different certificate?